### ISSUE TITLE
Add pillow and graphviz to doc build dependencies

### ISSUE THREAD

**Issue by dstansby** [matplotlib/matplotlib#7793]
Just spent about 5 mins building the docs, only to get:

```
Warning, treated as error:
/Users/dstansby/matplotlib/doc/examples/pylab_examples/agg_buffer.rst:8: WARNING: Exception occurred in plotting agg_buffer
 from /Users/dstansby/matplotlib/doc/mpl_examples/pylab_examples/agg_buffer.py:
Traceback (most recent call last):
  File "<string>", line 13, in <module>
ImportError: No module named 'PIL'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/dstansby/matplotlib/lib/matplotlib/sphinxext/plot_directive.py", line 525, in run_code
    six.exec_(code, ns)
  File "<string>", line 15, in <module>
SystemExit: Pillow must be installed to run this example

Building HTML failed.
```

Would be good to have pillow checked before the build starts like the other dependencies.

**Comment by tacaswell** [comment]
There is https://github.com/matplotlib/matplotlib/blob/master/doc-requirements.txt which has this.  I guess the docs have gotten out of date?

Should have the instruction to `pip install -r ../doc-requirements.txt` someplace.

**Comment by dstansby** [comment]
Also looks like the docs need `graphviz` to build too.

```
writing output... [ 58%] api/animation_api                                                                                                                                                                      
Warning, treated as error:
WARNING: dot command 'dot' cannot be run (needed for graphviz output), check the graphviz_dot setting

Building HTML failed.
```

**Comment by dstansby** [comment]
Huh, odd, it definitely checked for `numpydoc`, `ipython`, and `colorspacious` when I started trying to build (and force stopped the build at the beginning), but not `pillow` or `graphviz`.

**Comment by NelleV** [comment]
My bad… I forgot a couple of dependencies when I added the checks early on in the documentation build.
I'll fix that.

**Comment by dstansby** [comment]
Pretty sure this is fixed now.

**Comment by dstansby** [comment]
At least on OSX, `graphviz` is needed to build the docs, should this be added to the requirements?

```
reading sources... [100%] users/whats_new                                                                                                                
generating thumbnails... [100%] build/plot_directive/mpl_examples/pylab_examples/xcorr_demo.png                                                          
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [ 58%] api/animation_api                                                                                                               
Warning, treated as error:
WARNING: dot command 'dot' cannot be run (needed for graphviz output), check the graphviz_dot setting

Building HTML failed.
```

**Comment by dopplershift** [comment]
I didn't think graphviz was pip-installable. Is the requirement documented somewhere?

**Comment by NelleV** [comment]
graphvis is not pip installable.

**Comment by afvincent** [comment]
I can confirm that `graphviz` also seems to be required for building the docs on a GNU/Linux system (at least on Arch). I just got a nice build failure (the same as @dstansby ) after 10 min :/.

**Comment by afvincent** [comment]
Huh, how isn't `graphviz` pip-installable? I just genuinely tried `pip install graphviz` and I was returned a kind `Successfully installed graphviz-0.5.2` :). Let's see if the doc now builds.

For the record, I had run `pip install -r ../doc-requirements.txt` before getting the error about `graphviz`.

**Comment by dstansby** [comment]
https://pypi.python.org/pypi/graphviz/0.5.2

"To render the generated DOT source code, you also need to install Graphviz", so I guess there's no harm in adding this to `doc-requirements.txt`. Is there a way to test for graphviz before running the whole doc build though?

**Comment by tacaswell** [comment]
`which dot` will check if the underlying binary is installed or not.

**Comment by story645** [comment]
Closed in #8544 if not before.
